imap_processing/idex/idex_l1a.py

```diff
@@ -391,25 +445,21 @@ def _parse_high_sample_waveform(self, waveform_raw: str) -> list[int]:
         ints : list
             List of the high sample waveform.
         """
-        ints = []
-        for i in range(0, len(waveform_raw), 32):
-            # 32 bit chunks, divided up into 2, 10, 10, 10
-            # skip first two bits
-            ints += [
-                int(waveform_raw[i + 2 : i + 12], 2),
-                int(waveform_raw[i + 12 : i + 22], 2),
-                int(waveform_raw[i + 22 : i + 32], 2),
-            ]
-        return ints[:-4]  # Remove last 4 numbers
+        samples = self.MAX_HIGH_BLOCKS * self.NUMBER_SAMPLES_PER_HIGH_SAMPLE_BLOCK
+        if self.compressed.raw_value == 1:
+            ints: list[int] = rice_decode(
+                waveform_raw, nbit10=True, sample_count=samples
+            )
+            return ints[:-3]
```

Comment:
Out of curiosity, do you know why you need to put in `list[int]` as the type hint here since it looks like that is already defined in rice_decode as what it returns? 🤔 

```suggestion
            return rice_decode(waveform_raw, nbit10=True, sample_count=sample)[:-3]
```

Here you return from in the if-blocks, and down below you set a variable and then return. I'd suggest just being consistent between the two places and go with one version or the other.